chain/chunks/src/lib.rs

```diff
@@ -491,14 +495,68 @@ impl ShardsManager {
         Ok(())
     }
 
+    pub fn store_partial_encoded_chunk(
+        &mut self,
+        known_header: &BlockHeader,
+        partial_encoded_chunk: PartialEncodedChunk,
+    ) {
+        // Remove old partial_encoded_chunks
+        let encoded_chunks = &self.encoded_chunks;
+        self.stored_partial_encoded_chunks
+            .retain(|&height, _| encoded_chunks.height_within_front_horizon(height));
+
+        let header = partial_encoded_chunk
```

Comment:
Just return if it doesn't.
Expects are generally bad, but even worse because one can acutlly sent a message with no header, and now we need to analyze that all the callers actually reject it before we get here

Reply:
++